pythonbits/goodreads.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+from textwrap import dedent
+
+import goodreads_api_client as gr
+import pycountry
+
+from .config import config
+from .logging import log
+from .calibre import read_metadata
+from collections import OrderedDict
+
+config.register(
+    'Goodreads', 'api_key',
+    dedent("""\
+    To find your Goodreads API key, login to https://www.goodreads.com/api/keys
+    Enter the API Key below
+    API Key"""))
+
+EXCLUDED_WORDS = ['read', 'favorites', 'book',
+                  'own', 'series', 'novel', 'kindle', 'shelf'
+                  'library', 'buy', 'abandoned',
+                  'audible', 'audio', 'finish', 'wish']
```

Comment:
tbh i think it would be a better idea to have a whitelist instead, if there are so many shitty tags

Reply:
It's not too bad, I think an allow-list would be too hard as there's lots and lots of genres...

This is what comes back for Dune:

```xml
<popular_shelves>
<shelf name="to-read" count="498223"/>
<shelf name="currently-reading" count="54844"/>
<shelf name="science-fiction" count="13384"/>
<shelf name="sci-fi" count="12220"/>
<shelf name="favorites" count="8088"/>
<shelf name="fiction" count="6904"/>
<shelf name="fantasy" count="4244"/>
<shelf name="classics" count="3383"/>
<shelf name="scifi" count="2661"/>
<shelf name="owned" count="2454"/>
<shelf name="sci-fi-fantasy" count="1304"/>
<shelf name="books-i-own" count="974"/>
<shelf name="sf" count="937"/>
<shelf name="classic" count="789"/>
<shelf name="series" count="729"/>
<shelf name="audiobook" count="687"/>
<shelf name="dune" count="659"/>
<shelf name="scifi-fantasy" count="531"/>
<shelf name="space-opera" count="528"/>
<shelf name="fantasy-sci-fi" count="528"/>
<shelf name="audiobooks" count="505"/>
<shelf name="favourites" count="489"/>
<shelf name="novels" count="479"/>
<shelf name="kindle" count="468"/>
<shelf name="library" count="454"/>
<shelf name="default" count="443"/>
<shelf name="owned-books" count="395"/>
<shelf name="adventure" count="377"/>
<shelf name="to-buy" count="363"/>
<shelf name="dnf" count="344"/>
<shelf name="book-club" count="334"/>
<shelf name="science-fiction-fantasy" count="333"/>
<shelf name="abandoned" count="324"/>
<shelf name="audible" count="320"/>
<shelf name="adult" count="307"/>
<shelf name="audio" count="281"/>
<shelf name="speculative-fiction" count="273"/>
<shelf name="re-read" count="271"/>
<shelf name="space" count="264"/>
<shelf name="novel" count="252"/>
<shelf name="ebook" count="243"/>
<shelf name="did-not-finish" count="243"/>
<shelf name="my-library" count="237"/>
<shelf name="tbr" count="236"/>
<shelf name="literature" count="233"/>
<shelf name="epic" count="228"/>
<shelf name="great-american-read" count="219"/>
<shelf name="my-books" count="214"/>
<shelf name="sci-fi-and-fantasy" count="210"/>
<shelf name="books" count="207"/>
<shelf name="frank-herbert" count="207"/>
<shelf name="fantasy-scifi" count="206"/>
<shelf name="all-time-favorites" count="203"/>
<shelf name="unfinished" count="203"/>
<shelf name="science" count="196"/>
<shelf name="american" count="187"/>
<shelf name="on-hold" count="180"/>
<shelf name="20th-century" count="171"/>
<shelf name="english" count="160"/>
<shelf name="read-in-2020" count="158"/>
<shelf name="sff" count="158"/>
<shelf name="hugo-award" count="154"/>
<shelf name="favorite" count="154"/>
<shelf name="hugo" count="153"/>
<shelf name="sf-fantasy" count="153"/>
<shelf name="wish-list" count="152"/>
<shelf name="adult-fiction" count="146"/>
<shelf name="ebooks" count="144"/>
<shelf name="recommended" count="141"/>
<shelf name="dystopian" count="141"/>
<shelf name="reread" count="138"/>
<shelf name="favorite-books" count="138"/>
<shelf name="bookshelf" count="134"/>
<shelf name="fantascienza" count="132"/>
<shelf name="ciencia-ficción" count="130"/>
<shelf name="dystopia" count="129"/>
<shelf name="audio-books" count="127"/>
<shelf name="maybe" count="125"/>
<shelf name="religion" count="124"/>
<shelf name="read-2020" count="123"/>
<shelf name="i-own" count="123"/>
<shelf name="to-read-fiction" count="123"/>
<shelf name="must-read" count="118"/>
<shelf name="audio-book" count="115"/>
<shelf name="the-great-american-read" count="114"/>
<shelf name="to-re-read" count="112"/>
<shelf name="ciencia-ficcion" count="110"/>
<shelf name="nebula-award" count="110"/>
<shelf name="own-it" count="107"/>
<shelf name="to-reread" count="106"/>
<shelf name="owned-tbr" count="104"/>
<shelf name="hugo-winners" count="104"/>
<shelf name="home-library" count="103"/>
<shelf name="science-fiction-and-fantasy" count="103"/>
<shelf name="nebula" count="101"/>
<shelf name="physical-tbr" count="99"/>
<shelf name="paperback" count="99"/>
<shelf name="politics" count="99"/>
<shelf name="war" count="99"/>
<shelf name="sf-masterworks" count="98"/>
</popular_shelves>
```